include/ccf/indexing/strategies/seqnos_by_key_bucketed.h

```diff
@@ -0,0 +1,373 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/indexing/lfs_interface.h"
+#include "ccf/indexing/strategies/visit_each_entry_in_map.h"
+#include "ds/lru.h"
+
+namespace ccf::indexing::strategies
+{
+  // Stores only a subset of results in-memory, on-demand, and dumps the
+  // remainder to disk. The size of the per-key buckets which will be retained,
+  // and the number of buckets which may be held in-memory, are configurable
+  template <typename M>
+  class SeqnosByKey_Bucketed : public VisitEachEntryInMap
+  {
+  protected:
+    const size_t seqnos_per_bucket = 100;
```

Comment:
That seems like something that would be good to expose via configuration.

Reply:
I'll remove the default value here since it's unused. This is actually set in the constructor, so it is configurable by the app-developer. We don't currently have a mechanism to expose app-specific configuration in the per-node configuration, if that's what you mean (but would be a sensible thing to add, if we think heterogeneous/non-compile time node-configuration is necessary).